stdlib/public/TensorFlow/ShapedArray.swift

```diff
@@ -349,19 +349,15 @@ public extension ShapedArray {
     self.init(buffer: other.buffer, shape: other.shape)
   }
 
-  init(shape: [Int], scalars: [Scalar]) {
-    let scalarCount = shape.reduce(1, *)
-    precondition(scalarCount == scalars.count, "Scalar count mismatch.")
-    let buffer = TensorBuffer<Scalar>.create(count: scalarCount) { buffPtr in
-      let ptr = buffPtr.baseAddress!
-      scalars.withUnsafeBufferPointer { arrayBuf in
-        ptr.initialize(from: arrayBuf.baseAddress!, count: scalarCount)
-      }
-    }
+  init(shape: __owned [Int], scalars: __owned [Scalar]) {
+    precondition(shape.reduce(1, *) == scalars.count, "Scalar count mismatch.")
+    let buffer = TensorBuffer<Scalar>(allocation: .native(.init(scalars)),
```

Comment:
Nice removal of unnecessary buffer creation! That must be terrible for efficiency.